### ISSUE TITLE
Problem using my 8192-bit RSA keys

### ISSUE THREAD

**Issue by srleonbr** [open-keychain/open-keychain#2879]
I can't sign or encrypt with my 8192-bit RSA keys, I could in the previous version of the app.
When I use my EdDSA keys for signing and encryption, they work normally.

## Expected Behavior

When I updated the application, I expected that I would be able to use the same features as the previous version and with more new features, unfortunately I lost this function. It makes you want to go back to using the previous version.

## Current Behavior

Every time I try to sign or encrypt messages or files, a failure occurs that takes me to the home screen where all the keys are.
I want these keys to be compatible again as in the previous version.

## Possible Solution

I just wanted my 8192-bit RSA keys to work again. Since I'm not a developer, I don't know how this should be done.

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Context

I can no longer sign, encrypt or certify with my keys, they are important to me.

## Your Environment

* Android Version: 13
* Device Model: Motorola G52
* OpenKeychain Version: 6.0.0 (60000)
* From Google Play or F-Droid?: F-Droid

**Comment by Theta-Dev** [comment]
I experienced the same issue with version 6.0.0 trying to import 4096bit RSA keys from my PC. The older 5.8.2 works without issues.

**Comment by us4r01d** [comment]
I had the same Issue

**Comment by Valodim** [comment]
I was unable to reproduce this issue, both by adjusting our test suite to use rsa 8192 keys or importing them manually. not sure what's going on here, perhaps it's something different? :thinking: 

also, for the record, 8192 RSA keys are pointless, and should never have been supported by implementations.

**Comment by Valodim** [comment]
So I was unable to reproduce this issue, and hence can't currently fix it. Is anyone able to produce a secret key that is affected by this problem, that they could send to me (or just attach here)? That would be very helpful to find the problem

**Comment by Theta-Dev** [comment]
I have tested a bit more. Seems like the issue occurs when a key with a seperate subkey for signing is used. Here is the test key that I generated.

```
sec   rsa4096/D66BCF53F3FC27E5 2024-02-17 [SC]
      EEE4D36ADE94C31564E650FFD66BCF53F3FC27E5
uid                 [ultimate] John Doe <john.doe@example.com>
ssb   rsa4096/5B24DD6FB020C052 2024-02-17 [E]
ssb   rsa4096/F1364AD70DE65B99 2024-02-17 [S]
```

Steps to reproduce:
```
gpg --full-generate-key (with default settings)
gpg --edit-key <KEY_ID>
addkey
4 (RSA (sign only))
4096 (Keysize)
0 (Expiry)
save
```

Download testkey here (Password: 1234)

[openkeychain_testkey.txt](https://github.com/open-keychain/open-keychain/files/14319709/openkeychain_testkey.txt)

**Comment by rafaelazvdo** [comment]
> I have tested a bit more. Seems like the issue occurs when a key with a seperate subkey for signing is used. Here is the test key that I generated.
> 
> ```
> sec   rsa4096/D66BCF53F3FC27E5 2024-02-17 [SC]
>       EEE4D36ADE94C31564E650FFD66BCF53F3FC27E5
> uid                 [ultimate] John Doe <john.doe@example.com>
> ssb   rsa4096/5B24DD6FB020C052 2024-02-17 [E]
> ssb   rsa4096/F1364AD70DE65B99 2024-02-17 [S]
> ```
> 
> Steps to reproduce:
> 
> ```
> gpg --full-generate-key (with default settings)
> gpg --edit-key <KEY_ID>
> addkey
> 4 (RSA (sign only))
> 4096 (Keysize)
> 0 (Expiry)
> save
> ```
> 
> Download testkey here (Password: 1234)
> 
> [openkeychain_testkey.txt](https://github.com/open-keychain/open-keychain/files/14319709/openkeychain_testkey.txt)

This error is not limited only to RSA keys, NIST and ed/cv25519 keys also present this problem, and it occurs whenever there is a second subkey for signing, when trying to encrypt/sign a file, the application does not ask for the signing password and saves an invalid .pgp file that cannot be decrypted.

**Comment by rafaelazvdo** [comment]
@srleonbr What is the structure of your key, How many subkeys and what is their function?

**Comment by Valodim** [comment]
I should be able to work with that, thanks @rafaelazvdo :)

**Comment by Valodim** [comment]
Got it, please try with https://github.com/open-keychain/open-keychain/commit/5cc2d88ab9dff0c7f4bc5e8575ee9f860470228f applied. I'll make a 6.0.2 then

**Comment by srleonbr** [comment]
> So I was unable to reproduce this issue, and hence can't currently fix it. Is anyone able to produce a secret key that is affected by this problem, that they could send to me (or just attach here)? That would be very helpful to find the problem

Sorry for taking so long to respond.
I had to generate another key, I was forced to revoke the others I had in order to make my daily subscriptions.
https://send.portailpro.net/download/55e9edff91f0202f/#tMCdvo4hVQG41ACEuWph2A
Even if you have already decided to make a new version, it would be interesting to test this one, because you tested it on a 4096-bit RSA key, this key is 8192-bit RSA.

**Comment by srleonbr** [comment]
> @srleonbr What is the structure of your key, How many subkeys and what is their function?

There are three 8192-bit RSA subkeys, all of which authenticate, sign and encrypt, but only the main one certifies.

You can check this key I recently created identical to my personal one.
https://send.portailpro.net/download/55e9edff91f0202f/#tMCdvo4hVQG41ACEuWph2A

**Comment by Valodim** [comment]
> There are three 8192-bit RSA subkeys, all of which authenticate, sign and encrypt, but only the main one certifies.

Thanks for the reproducerI wonder what the point of such a key is. :thinking: 

Regardless, the RSA bitsize didn't actually cause the issue described here, but rather the multiple signing key flags (which is slightly unusual but perfectly fine to have)

**Comment by Valodim** [comment]
I added some unit tests to nail down this behavior, and released 6.0.2
